### PR TITLE
rustbuild: drop tool::should_install

### LINKED ISSUE #74431: Rust 1.45.0 and above no longer install clippy
When updating from 1.44.1 to 1.45.0, clippy-driver and cargo-clippy are no longer installed by `./x.py install`.
`miri` and `cargo-miri` weren't previously installed but they are now, and the opposite is true for clippy-driver and cargo-clippy

### PR DESCRIPTION
**Pull request by Keruspe**
Always install when the build succeeds

Fixes #74431

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
@bors r+ rollup

I'm going to go ahead and beta-nom and accept this as well -- it's a harmless change IMO and really only affects consumers who are directly using x.py.

File: src/bootstrap/install.rs [hunk 1]
```diff
@@ -221,27 +215,14 @@ install!((self, builder, _config),
...
     RustAnalyzer, "rust-analyzer", Self::should_build(_config), only_hosts: true, {
         builder.ensure(dist::RustAnalyzer { compiler: self.compiler, target: self.target });
-        if Self::should_install(builder) {
-            install_rust_analyzer(builder, self.compiler.stage, self.target);
-        } else {
-            builder.info(
-                &format!("skipping Install rust-analyzer stage{} ({})", self.compiler.stage, self.target),
-            );
-        }
+        install_rust_analyzer(builder, self.compiler.stage, self.target);
```
Comment (mati865) [inline src/bootstrap/install.rs:218]: AFAIK rust-analyzer is not guaranteed to always build successfully. Therefore install should be performed only when build succeeds.

Comment (Mark-Simulacrum) [reply src/bootstrap/install.rs:218]: rust-analyzer builds on stable today, so is indeed expected to always build successfully.

**Comment by Keruspe** [comment]
@Mark-Simulacrum Is there any chance to get this in 1.45.1? (A plain "no" is fine by me, especially since it's not essential and the patch won't apply cleanly due to rust-analyzer, but since there'll be a 1.45.1 I figured why not?)

**Comment by Mark-Simulacrum** [comment]
Yeah, I'm fine accepting this for stable backport as well.

**Pull request merged**